plugins/gpg-agent/gpg-agent.plugin.zsh

```diff
@@ -1,14 +1,16 @@
-# Enable gpg-agent if it is not running
-GPG_AGENT_SOCKET="${XDG_RUNTIME_DIR}/gnupg/S.gpg-agent.ssh"
-if [ ! -S $GPG_AGENT_SOCKET ]; then
-  gpg-agent --daemon >/dev/null 2>&1
-  export GPG_TTY=$(tty)
+# Enable gpg-agent if it is not running-
+# --use-standard-socket will work from version 2 upwards
+
+AGENT_SOCK=`gpgconf --list-dirs | grep agent-socket | cut -d : -f 2`
+
+if [ ! -S ${AGENT_SOCK} ]; then
+  gpg-agent --daemon --use-standard-socket >/dev/null 2>&1
 fi
+export GPG_TTY=$(tty)
```

Comment:
This can be `$TTY` directly, no need for an extra fork.

Reply:
No problem